### PR TITLE
Update contributor documentation

### PR DESCRIPTION
**Pull request by rye**
I think it's better if this is more of an actual list of things to do rather than just notes, which seemed to be the old style. Kinda WIP, but can get merged whenever.

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -1,15 +1,12 @@
...
+Would you like to contribute?
+Great!
+Have a look at [React Native's "getting started" guide][rn-gs] to help you get your computer set up.
+Then, fork the repository, and make a pull request!
 
-If you are looking to get started with the project, issues tagged with
-[`good first issue`][gfi] are a great place to start. These issues are
-ones that the team has identified as simple, small changes that won't
-get too deep, and will be easy to approve for merging.
+If you are looking to get started with the project, issues tagged with [`good first issue`][gfi] are a great place to start.
```
Comment (drewvolz) [inline CONTRIBUTING.md:8]: Looks like we run out of those types of issues from time to time (like now!). Should we also note that the rest of our issues need help, too?

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -22,81 +19,51 @@ get too deep, and will be easy to approve for merging.
...
-- `npm run prettier`: Prettier enforces a common style on the JS code,
-  without us needing to edit anything
+- `npm run prettier`: Prettier enforces a common style on the JS code, without us needing to edit anything
 
-We have a special command that runs all four of those for you: `npm
-run check`.  So, before you commit and push, you may wish to run `npm
-run check`.  If you don't, TravisCI will run them for you and our
-StoDevX Bot will comment in the PR to let you know exactly what
-happened.
+We have a special command that runs all four of those for you: `npm run check`.
```
Comment (drewvolz) [inline CONTRIBUTING.md]: This needs to be removed since we removed the `check` command from package.json.

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -109,18 +76,28 @@ As always, please keep the [Code of Conduct][cc] in mind.
...
+(`-beta.W` where `W` is a monotonically increasing number.)
+Technically this can be whatever you want.
+
+The bare minimum: `npm version <version>`, e.g. `npm version 2.7.0`.
+It is probably a good idea to run a beta before doing a full release, to ensure that all kinks in the uploading pipeline are worked out.
+This also lets you distribute beta builds to beta testers.
+
+#### Release Commit Style
+
+@rye prefers this style _a lot_ to the `npm version` style.
```
Comment (drewvolz) [inline CONTRIBUTING.md]: 😆

File: CONTRIBUTING.md (hunk 1 above)
Comment (rye) [reply CONTRIBUTING.md:8]: Yeah, I'll add that.

**Review by drewvolz — APPROVED** [review]

**Comment by rye** [comment]
I'll ship this for now and continue playing around with it as I have time.

**Pull request merged**